pkg/roll/update_test.go

```diff
@@ -688,3 +689,83 @@ func TestShouldRollMidwayHealthyMigrationFromZeroWhenNewSatisfies(t *testing.T)
 	Assert(t).AreEqual(remove, 0, "expected to remove no nodes")
 	Assert(t).AreEqual(add, 1, "expected to add one node")
 }
+
+func panicOnError(desc string, errs <-chan error) {
+	err, ok := <-errs
+	if ok {
+		panic("Error received on " + desc + ":" + err.Error())
+	}
+}
+
+// Transfers the named node from the old RC to the new RC
+func transferNode(node string, manifest pods.Manifest, upd update) error {
+	if _, err := upd.kps.SetPod(kp.REALITY_TREE, node, manifest); err != nil {
+		return err
+	}
+	podID := manifest.ID().String()
+	return upd.labeler.SetLabel(labels.POD, node+"/"+podID, rc.RCIDLabel, string(upd.NewRC))
+}
+
+func TestRollLoopTypicalCase(t *testing.T) {
+	upd, _, manifest := updateWithHealth(t, 3, 0, nil, nil, nil)
+	upd.DesiredReplicas = 3
+	upd.MinimumReplicas = 2
+
+	updates := make(chan map[string]health.Result)
```

Comment:
this name is... slightly misleading, since there's an `update` type, but this channel is not for values of that `update` type. I meant for it to be "health updates". I'll probably change the name. `health` doesn't work since it's the package name and I use `health.Passing` later on. so probably `healthChan` or something.